### PR TITLE
[luajit] Remove dllexport in static build

### LINKED ISSUE #26511: [luajit] nearly all CRT functions from stdio are exported by lua51.lib
Windows 11 x64 + VS2019 x64 build, luajit:x64-windows-static.

If I statically linking to lua51.lib, my dll/exe export table will contain 170+ CRT functions such as "sprintf/sscanf/fprintf" etc.

modify src\msvcbuild.bat,

change "/D_CRT_STDIO_INLINE=__declspec(dllexport)__inline" to "/D_CRT_STDIO_INLINE=__inline"

### PR DESCRIPTION
**Pull request by LilyWangLL**
**Describe the pull request**

- #### What does your PR fix?
  Fixes #26511, remove `dllexport` in static build.

### CODE REVIEW

File: ports/luajit/portfile.cmake [hunk 1]
```diff
@@ -13,6 +13,7 @@ if (VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
     set (LJIT_STATIC "")
 else()
     set (LJIT_STATIC "static")
+    vcpkg_replace_string("${SOURCE_PATH}/src/msvcbuild.bat" "/D_CRT_STDIO_INLINE=__declspec(dllexport)__inline" "/D_CRT_STDIO_INLINE=__inline")
```
Comment (JackBoosY) [inline ports/luajit/portfile.cmake:16]: This need the upstream confirm.

**Comment by LilyWangLL** [comment]
Closing this PR because of this change will cause other usage problems, see comment: https://github.com/LuaJIT/LuaJIT/issues/889#issuecomment-1229169333.

**Comment by z16166** [comment]
so this problem can't be solved?

**Comment by LilyWangLL** [comment]
> so this problem can't be solved?

This problem is upstream issue, this need upstream fix. If we just change the source file as following this PR, it will cause other problems like manager MikePall says.

**Pull request closed without merging**